frontend/src/components/import-button.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { Button, Dropdown, DropdownButton, Modal } from "react-bootstrap";
+import { readFile } from "../libs/fileManager";
+
+export class ImportButton extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: null,
+            dialogContents: "",
+            dialogOpen: false
+        };
+    }
+
+    /* assume frontend data is up to date
+     *  upload the assignment object
+     *  if there's an part of inconsistency between the imported data and frontend data
+     *  then apiGET that part of data and re-verify it
+     */
+
+    handleClose = () => {
+        this.setState({ dialogOpen: false });
+    };
+
+    uploadDataToBackend = () => {
+        // TODO
+    };
+
+    importFile = () => {
+        let loadDataFunc = data => {
+            // passed in data is of json format
+            console.log(data);
+
+            /* TODO: 
+            * const diffs = getDiffs(data, ...dataFromBackend);
+            * if (diffs) {
+            *     this.setState({ data: data, dialogContents: diffs, dialogOpen: true})
+            * } else {
+            *     uploadDataToBackend(data)
+            } */
+        };
+
+        readFile(document.getElementById("file-input"), loadDataFunc);
+    };
+
+    render() {
+        return (
+            <div>
+                <DropdownButton id="dropdown-basic-button" title="Import">
+                    <Dropdown.Item
+                        onClick={() =>
+                            document.getElementById("file-input").click()
```

Comment:
This is a no no! If your code has a `document.getElementById`, you're probably doing something wrong. See here https://github.com/siefkenj/utk/blob/fe962f0b8c09fe386fc170caaedad5510a0946cb/src/components/savers.js#L45 for how you can implement something appropriate. (It involves setting a `label` for the `input`)